### DISCUSSION TITLE
QR code authentication using frejaeidclient

### DISCUSSION THREAD

**Discussion by Djawahl** [Freja-eID/frejaeidclient#95] [Q&A] [answered]
I have a question regarding QR code authentication using frejaeidclient. The following steps are included in the process according to the documentation from Freja:
---
1. Create initAuthRequest with "userInfoType":"INFERRED" and "userInfo":"N/A".
This first step can be done using frejaeidclient.
public String initiateAuthenticationRequestCustomUsingQRCode(
    AuthenticationClientApi authenticationClient) throws FrejaEidClientInternalException, FrejaEidException {
    UserInfoBuilder userInfoBuilder = InitiateAuthenticationRequest.createCustom();
    SetOptionalParamsBuilder setOptionalParamsBuilder = userInfoBuilder.setInferred();
    setOptionalParamsBuilder.setAttributesToReturn(
        AttributeToReturn.SSN,
        AttributeToReturn.BASIC_USER_INFO,
        AttributeToReturn.EMAIL_ADDRESS);
    setOptionalParamsBuilder.setMinRegistrationLevel(MinRegistrationLevel.PLUS);
    InitiateAuthenticationRequest request = setOptionalParamsBuilder.build();
    return authenticationClient.initiate(request);
}

2. Perform URL encoding on the authRef received in the response.
3. Insert the result into the following schema: frejaeid://bindUserToTransaction?transactionReference=
4. Perform URL encoding on the previous schema.
5. Use this as the parameter for the HTTP GET method which generates Freja eID QR code:
The following end point is not yet implemented in this library?
/qrcode/generate?qrcodedata=frejaeid%3A%2F%2FbindUserToTransaction%3FtransactionReference%3DW5IJKBH95h9KRhICimRia5otxkVuP36Aw5CRLYOteIQfx4bpfmvDnW7AbDugMjIa

6. Display the generated QR code as an image on your web page for the user to scan.
---

Are there any plans to implement this in the near future?

**Comment by IgorGnusFreja** [comment, marked as answer]
Hello,
 Thank you for your question and suggested feature. We are happy to announce that we have implemented this feature in our latest client version - 2.17.0. Please see our updated documentation for more information, under  https://frejaeid.atlassian.net/wiki/spaces/DOC/pages/8847365/Authentication+Service+Client#Generate-Authentication-QR-Code 
 
Kind regards,  
FrejaEid development team
